data-collection/deploy/module-budgets.yaml

```diff
@@ -128,6 +129,27 @@ Resources:
                       return o.isoformat()
                   return None
 
+          def clean_value(value):
+              """Clean string values by replacing special characters with underscores"""
+              if isinstance(value, str):
+                  return re.sub(r'[:\\]', '_', value)
+              elif isinstance(value, list):
+                  return [clean_value(v) for v in value]
+              return value
+
+          def process_cost_filters(budget):
+              """Process and clean cost filters in budget"""
+              cost_filters = budget.get('CostFilters', {})
+              if not cost_filters:
+                  budget['CostFilters'] = {'Filter': ['None']}
+                  return
+
+              cleaned_filters = {
+                  re.sub(r'[:\\]', '_', key): clean_value(value)
```

Comment:
i would go with all non alpanumeric

Reply:
I have pushed the latest updates with '\W+' to cover all special characters.